README.md

```diff
@@ -1 +1,94 @@
-Placeholder for Checkstyle Regression Tool readme.
+# Regression-tool
+
+Regression-tool is an automation tool to do regression testing based on proposed patch(Pull Request) 
+of checkstyle project. The tool could generate configuration based on Git changes, which is used to 
+generate diff report between the base and the patch branch.
+
+## Setup
+
+### Requirements
+
+- Checkstyle repository which including both master and patch branch need to be cloned.
+In most case, it should be your forked repository.
+- [Contribution repository](https://github.com/checkstyle/contribution) need to be cloned.
+- Regression-tool uses environment variable `M2_HOME` or system property `maven.home` to find the maven binary files.
+You need to specify either `M2_HOME` or `maven.home` in advance. You could get the path of maven home by running `mvn -v`.
+
+### Clone
+
+Just clone this repository to your local.
+
+```bash
+$ git clone git@github.com:checkstyle/regression-tool.git
+```
+
+Or use HTTPS mode.
+```bash
+$ git clone https://github.com/checkstyle/regression-tool.git
+```
+
+### Generate Jar
+
+Generate a `regression-tool-XX-all.jar`(XX means the version of regression-tool) in the `target` folder.
+
+```bash
+$ cd /path/to/regression-tool
+$ mvn clean package -Passembly
+```
+
+## Usage
+
+```bash
+$ java -jar regression-tool-XX-all.jar -p <arg> -r <arg> [--stopAfterGenerateConfig] [-t <arg>]
+```
+
+### Arguments
+
+#### localGitRepo (r)
+
+The path of the checkstyle repository. **Required, one argument.**
+
+#### patchBranch (p)
+
+The name of the PR branch, which would be compared with the master. **Required, one argument.**
+
+#### stopAfterGenerateConfig
+
+Indicates that regression-tool would stop after generating config. By default, the tool would generate 
+the diff report after generating config. **Optional, no argument.**
+
+The report generation might requires network and be time-consuming. 
+If you are not able to connect to network or just don't want to do the generation right now, 
+you could use this mode.
+
+
+#### checkstyleTesterPath (t)
+
+The path of the checkstyle-tester directory. If you are not using `--stopAfterGenerateConfig` mode, this 
+option is required, otherwise this could be absent. **Optional, one argument.**
```

Comment:
@romani This is optional because `tester` repo isn't needed to generate configuration. It is only needed to run configuration against repo.
As description states:
>  If you are not using `--stopAfterGenerateConfig` mode, this option is required